Update to Yocto 5.3 - Whinlatter

hnez opened the pull request:
This is a very early draft that barely compiles yet.

TODO before merging:

- [x] Clean up into usable commits.
- [x] Check every file we have to make sure it is still needed.
- [x] Test on hardware.
- [x] Update the whinlatter dummy branch to the most recent walnascar state.
- [x] Rebase this PR on top of it.

hnez committed:
meta-lxatac-software: ripgrep: update to 15.1.0

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: python3-usbsdmux: drop no longer required bbappend

The service was not used for many years and is no longer installed in
recent usbsdmux versions.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
ci: build: update for whinlatter branch

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: distro: tacos: change release name to whinlatter

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: tacd-webinterface: do not hardcode git dir

And adapt to UNPACKDIR changes.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez commented:
Reminder: use `go-mod-update-modules.bbclass`.

hnez committed:
meta-lxatac-software: gitlab-runner: upgrade and use update-modules

Use recipetool to re-generate the recipe using
`inherit go-mod-update-modules`.

This allows us to finally get rid of `do_compile[network] = "1"`.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: github-act-runner: upgrade and use update-modules

Use recipetool to re-generate the recipe using
`inherit go-mod-update-modules`.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-*,bitbake,oe-core: update to recent whinlatter/master branches

This also means getting rid of the poky submodule, since the poky
repository - which pre packaged openembedded-core and bitbake
together - is no longer being updated.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-{bsp,software}: update layer.conf for whinlatter

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-{bsp,software}: transition from WORKDIR to UNPACKDIR

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: distro: tacos: do not enable icecc support

The icecc class was removed in the whinlatter release.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: remove `S = "${WORKDIR}/git"` from recipes

This is the default for git based recipes and an error is raised if it is
set anyways.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: python3-pycares: remove backported version

The version in meta-oe is now 4.11 and thus new enough to no longer
cause issues for our build.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: bottom: update to 0.12.3

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: add tag information to release recipes using git

The tag is checked against the commit hash since whinlatter.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: python3-lxa-iobus: work around native http parser

… not working for us. Setting `AIOHTTP_NO_EXTENSIONS=1` results in the
pure python HTML parser being used instead of the native C version.

The C version started failing for us since switching from walnascar to
whinlatter with the following error in the web interface:

    Invalid header value char:

      b'Host: lxatac-00011:8080'
                              ^

And the following Error on the command line:

    ERROR:aiohttp.server:Error handling request from 192.168.94.2
    Traceback (most recent call last):
      File "/usr/lib/python3.13/site-packages/aiohttp/web_protocol.py", line 408, in data_received
        messages, upgraded, tail = self._request_parser.feed_data(data)
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
      File "aiohttp/_http_parser.pyx", line 558, in aiohttp._http_parser.HttpParser.feed_data
    aiohttp.http_exceptions.BadHttpMessage: 400, message:
      Invalid header value char:

        b'Host: lxatac-00011:8080'
                                ^
    ERROR:aiohttp.server:Error handling request from 192.168.94.2
    Traceback (most recent call last):
      File "/usr/lib/python3.13/site-packages/aiohttp/web_protocol.py", line 408, in data_received
        messages, upgraded, tail = self._request_parser.feed_data(data)
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
      File "aiohttp/_http_parser.pyx", line 558, in aiohttp._http_parser.HttpParser.feed_data
    aiohttp.http_exceptions.BadHttpMessage: 400, message:
      Invalid header value char:

        b'Host: lxatac-00011:8080'
                                ^

Ideally we would find out what is wrong instead of working around the
symptoms, but that involves a lot of layers of software.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez commented:
I think this is finally ready for review. If you want to test a build of this on your TAC run:

```
$ rauc-enable-cert devel.cert.pem
$ rauc install https://github-runner.pengutronix.de/artifacts/forrest-build-4Yu3DlIMy5rLLZvT/lxatac-core-bundle-base-lxatac.raucb
```

jluebbe commented on .gitmodules:
```diff
@@ -1,32 +1,40 @@
+[submodule "meta-yocto"]
+	path = meta-yocto
+	url = https://git.yoctoproject.org/meta-yocto
+	branch = whinlatter
```
This seems to be for meta-poky, but as far as I can see, we don't really use anything useful from there.

ejoerns replied:
I also don't think that you need `meta-yocto`. It includes `meta-poky` which just provides the `poky` example distro (and variants) and `meta-yocto-bsp` which provides the example `generic*` machines (and beagle play).

The reason it is in is probably `poky-sanity.bbclass`. I'd drop this instead from the distro since it just adds the poky-specific check for `REQUIRED_POKY_BBLAYERS_CONF_VERSION`. If that mechanism is found useful, it could also be imported (and renamed).

hnez replied:
Thanks for catching this. I have removed `meta-yocto` now :+1:.

hnez committed:
meta-lxatac-software: github-act-runner: set SPDX where previously unknown

The `go-mod-update-modules.bbclass` luckily parses the previous
`-licenses.inc` for SPDX identifier and license file hashes,
so doing this will not be in vain when the next update comes around
(even though the files says that all content will be replaced).

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: gitlab-runner: set SPDX and update_modules

The `go-mod-update-modules.bbclass` luckily parses the previous
`-licenses.inc` for SPDX identifier and license file hashes,
so doing this will not be in vain when the next update comes around
(even though the files says that all content will be replaced).

Just re-running `bitbake -c update_modules gitlab-runner` also
prunes go modules that are included in the `go.mods` but not
actually used in the project.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-yocto: remove

We only dependent on the empty `poky-world-exclude.inc` and the
poky-specific `poky-sanity.bbclass`.

Since we do not really _need_ these, we can get rid of meta-yocto
altogether.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez commented:
Soo … are we good to go?

hnez commented:
> Soo … are we good to go?

Sorry for bringing this up again, but are we?
